indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -524,7 +523,9 @@ public List<SegmentAllocateResult> allocateSegments(
 
   /**
    * Marks the segment allocation as failed if the underlying task is not active.
+   * Should be called only while holding the {@code giant} lock.
    */
+  @GuardedBy("giant")
```

Comment:
We should probably also annotate the field `activeTasks` with `@GuardedBy("giant")`.

Reply:
No, I meant the `private final Set<String> activeTasks = new HashSet<>();` inside `TaskLockbox` class should be annotated with guarded by.
The only uncovered path would be `getActiveTasks()` which we should fix up.

Reply:
Ah, I see. Thanks for the clarification. For some reason, my IDE was not flagging this one.
We can skip adding the annotation on `activeTasks` for the time being then since having the two try-finally doesn't seem clean.